src/index.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import injectSheetDefault from 'react-jss'
+
+
+type StyledElementType = Function & { Tag: string, styles: Object }
+type PrimitiveProps = {
+  classes: Object,
+  children?: any,
+  className?: string,
+}
+
+
+export const prepareStyled = (injectSheet?: Function = injectSheetDefault) =>
+  (
+    Element: string | StyledElementType,
+    styles: Object,
+    baseStyles?: Object = {},
+  ): StyledElementType => {
+    const {
+      Tag,
+      styles: inheritStyles = {},
+    }: {
+      Tag: string,
+      styles?: Object
+    } = typeof Element === 'string' ? { Tag: Element } : Element
+
+    const elementStyles = { ...inheritStyles, ...styles }
+
+    const Primitive = ({ classes, children, className }: PrimitiveProps) =>
+      <Tag className={classes[Tag].concat(className ? ` ${className}` : '')}>
```

Comment:
how is it possible that you use `Tag` as a component and as a class name at the same time?